extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapStore.java

```diff
@@ -386,39 +289,20 @@ private GenericRecord convertRowToGenericRecord(K key, SqlRow row) {
     public Map<K, GenericRecord> loadAll(Collection<K> keys) {
         awaitInitFinished();
 
-        Object[] keysArray = keys.toArray();
+        return super.loadAll(keys);
 
-        try (SqlResult queryResult = sql.execute(queries.loadAll(keys.size()), keysArray)) {
-            Iterator<SqlRow> it = queryResult.iterator();
-
-            Map<K, GenericRecord> result = new HashMap<>();
-            while (it.hasNext()) {
-                SqlRow row = it.next();
-                K id = row.getObject(properties.idColumn);
-                GenericRecord record = convertRowToGenericRecord(id, row);
-                result.put(id, record);
-            }
-            return result;
-        }
```

Comment:
We shouldn't need to override the load* methods in the `GenericMapStore` class.

Reply:
GenericMapStore inherits from GenericMapLoader